core/src/jac.jl

```diff
@@ -391,18 +390,40 @@ function formulate_jac!(
             continue
         end
 
-        id_pump = only(outneighbors(graph_control, id))
+        controlled_node_id = only(outneighbors(graph_control, id))
+        controls_pump = (controlled_node_id in pump.node_id)
```

Comment:
The brackets can be removed here. But looking at the if-else block below, it'd be better to replace the else with a check if it is a weir, and add an else that throws an error, to be a bit more defensive.

And rather than checking all pump nodes with `x in y`, you could use `isempty(searchsorted([2,3], 4))`, which we could perhaps add as a utility function besides `findsorted`, with a name like `insorted`. Ha, actually I just found out that `insorted` already exists in Base.

Reply:
I added an extra check. In a case like `controls_pump = (controlled_node_id in pump.node_id)` I do prefer to write the brackets otherwise I find the syntax confusing.